### ISSUE TITLE
RichApi.Error with content control when changing properties using set or insertHtml

### ISSUE THREAD

**Issue by kgeorgiou1234** [OfficeDev/office-js#2815]
## Expected Behavior
Able to insert html in content control multiple times in Word online and update it's properties, for example title, even when the user's cursor is in the content control.

## Current Behavior
When the user clicks in the content control and types something leaving the cursor in the content control and then code tries to modify the content control HTML, there are errors thrown.

Currently there are two errors thrown
1. When in the code we use only insertHtml, then the error thrown is "La is not iterable"
2. When in the code we use clear and then insertHtml (suggested in another ticket), then the error thrown is "Sys.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. Parameter name: index"

## Steps to Reproduce, or Live Example
Code attached in files 'Script_NotIterable.txt' and 'Script_Index.txt' in order to use in Script Lab
1. Open word online, load Script lab and add the script from 'Script.txt'
2. Run the script
3. Click 'Setup', and then 'Insert'
4. Click 'Modify content controls'
5. Type something in the second content control (green color)
6. Click 'Modify content controls' and you will get the error "La is not iterable" (file 'Error from console - insertHtml.txt')

Go back to the code of ScriptLab, uncomment line 61 (oddContentControls.items[j].clear();) and run it again
1. Click 'Setup' and then 'Insert'
2. Click 'Modify content controls', the first time it works
3. Click 'Modify content controls' again, here it will give an error for index (file 'Error from console - clear and insertHtml - second modify with cursor in second content control.txt')

## Context
Users in our add-in load the HTML from the database, they can then edit it in the content control and when they are happy with the final HTML they can save it back to the database.

## Your Environment

* Platform [PC desktop, Mac, iOS, Office Online]: Office Online
* Host [Excel, Word, PowerPoint, etc.]: Word
* Operating System: Windows
* Browser (if using Office Online): Edge

## Useful logs

[Script.txt](https://github.com/OfficeDev/office-js/files/9497054/Script.txt)
[Error from console - clear and insertHtml - second modify with cursor in second content control.txt](https://github.com/OfficeDev/office-js/files/9497055/Error.from.console.-.clear.and.insertHtml.-.second.modify.with.cursor.in.second.content.control.txt)
[Error from console - insertHtml.txt](https://github.com/OfficeDev/office-js/files/9497056/Error.from.console.-.insertHtml.txt)

**Comment by JinghuiMS** [comment]
Thanks for reporting this issue. It's definitely not right to have this error, but fortunately there is a workaround to unblock us. It has been put on our backlog<ADO#6649732> for internal track. We will keep track of this issue and let you know if there are any updates.

**Comment by kgeorgiou1234** [comment]
So what is the workaround?

**Comment by nickamckenna** [comment]
We are also experiencing this problem. @JinghuiMS What workaround did you find?

**Comment by ubabukos** [comment]
Hi @JinghuiMS I'm also experiencing this issue with the `ContentControl.insertHtml()` method. Any update on why this is happening? Are you able to share the workaround with us too? 

Below is the debug info I get from the Office.js API

```JSON
{
    "wordAPICode": "GeneralException",
    "errorLocation": "ContentControl.insertHtml",
    "innerError": "",
    "wordAPIMessage": "Sys.ArgumentOutOfRangeException: Specified argument was out of the range of valid values.\nParameter name: index",
    "statements": "var insertHtml2 = v4.insertHtml(...);",
    "surroundingStatements": "...\nv2.appearance = ...;\nvar v3 = context.root._getObjectByReferenceId(\"{98ab96e9-46a3-4c5c-bf5c-b6070e9b3667}{179}\");\nvar insertHtml1 = v3.insertHtml(...);\n// Instantiate {insertHtml1}\nvar v4 = context.root._getObjectByReferenceId(\"{98ab96e9-46a3-4c5c-bf5c-b6070e9b3667}{157}\");\n// >>>>>\nvar insertHtml2 = v4.insertHtml(...);\n// <<<<<\n// Instantiate {insertHtml2}"
}
```

**Comment by kgeorgiou1234** [comment]
We have not received any workarounds, still waiting for that or a fix.

**Comment by jingxuan0109** [comment]
@ubabukos @kgeorgiou1234 @nickamckenna 
For scenario 1, I could not reproduce. I am not sure whether this is because step 5. For step 5, I typed something at the end of content control. Could you help me to reproduce on first scenario?

**Comment by kgeorgiou1234** [comment]
@jingxuan0109 you need to have a new document for each scenario. It is essentially the same thing as scenario 2.
What do you need for the reproduce? 
By the way, is it anywhere close to being resolved?
Please note that we use sensitivity labels for the documents.

**Comment by jingxuan0109** [comment]
Hi @kgeorgiou1234, For the first scenario, I created a new document and try the same steps. It works well.

**Comment by kgeorgiou1234** [comment]
@jingxuan0109 seems like it is working now, thank you!
